WeatherApp/BL/WeatherHistoryService.cs

```diff
@@ -0,0 +1,60 @@
+﻿using BL.Models;
+using DAL;
+using DAL.Models;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BL
+{
+    public class WeatherHistoryService : IWeatherHistoryService
+    {
+        private readonly ILogger _logger;
+        private readonly IWeatherHistoryRepository _weatherHistoryRepository;
+        private readonly IWeatherService _weatherService;
+
+        public WeatherHistoryService(ILogger<WeatherHistoryService> logger, IWeatherHistoryRepository weatherHistoryRepository, IWeatherService weatherService) 
+        {
+            _logger = logger;
+            _weatherHistoryRepository = weatherHistoryRepository;
+            _weatherService = weatherService;
+        }
+
+        public async Task UpdateWeatherHistory(IEnumerable<string> cityNames)
```

Comment:
try to use params string[]

Reply:
Params string[] is now used for cityNames.